hephaistus/main-theme/distribution-system/chord/playground/nullist0/lib/src/main/kotlin/org/lia/dht/nullist0/DistributedHashTableService.kt

```diff
@@ -0,0 +1,13 @@
+package org.lia.dht.nullist0
+
+import java.net.URI
+
+/**
+ * An interface for distributed hash table(DHT) service. To get some [DistributedHashTable] instance,
+ * client should call [create] function, or [join] function to some running DHT uri information.
+ */
+interface DistributedHashTableService<K, V> {
+    fun create(port: Short): DistributedHashTable<K, V>
+
+    fun join(uri: URI): DistributedHashTable<K, V>
```

Comment:
Chord is p2p DHT which means it does not have any leader or manager to maintain its cluster. So, `ChordDistributedHashTable` need to have create / join. How do you think?

Reply:
As you mentioned, the p2p related concerns will be covered in `ChordDistributedHashTable`. But this separation didn't come from the server-client notion. This is come from the following intuition. IMO, the common use case of `DistributedHashTable` is the followings.

1. Create **new** DHT(Chord ring) or join **created** DHT(Chord ring).
2. Get or set some key-values to DHT (Chord ring).
3. Release or close DHT (Chord ring).

The most of usage will be **the number 2**, and this will be used more than once. Also, these functionalities have different notions of Steps.

1. Configuration Step: Create **new** DHT(Chord ring) or join **created** DHT(Chord ring).
2. Usage Step: Get or set some key-values to DHT (Chord ring).
3. Finalize Step: Release or close DHT (Chord ring).

Thus I hope to separate the `create` / `join` from the `ChordDistributedHashTable`. If there is a problem, please let me know.

Reply:
There is no problem, I want to ask your intention. Thank you 👍